comms/discovery/rpcz/apply.go

```diff
@@ -152,22 +152,35 @@ func (proc *RPCProcessor) Apply(rpcLog *schema.RpcLog) error {
 		return nil
 	}
 
-	userId, err := queries.GetUserIDFromWallet(db.Conn, context.Background(), wallet)
-	if err != nil {
-		logger.Warn("wallet not found: "+err.Error(), "wallet", wallet, "sig", signatureHeader)
-		return nil
+	var userId int32
+
+	// attempt to read the (newly added) current_user_id field
+	if rawRpc.CurrentUserID != "" {
+		if u, err := misc.DecodeHashId(rawRpc.CurrentUserID); err == nil && u > 0 {
+			// valid current_user_id + wallet combo?
+			// for now just check that the pair exists in the user table
+			// in the future this can check a "grants" table that a given operation is permitted
```

Comment:
cc @nicoback2 re: delegated DMs work - I'm guessing these tables already exist? Could probably have this wired up if you wanted